### PR TITLE
Social: Add Jetpack Social Basic to My Products

### PR DESCRIPTION
**Pull request by spsiddarthan**
Jetpack Social Products weren't showing up under the my products section in the Jetpack Dashboard. 

#### Changes proposed in this Pull Request:

* Made the changes to list Jetpack Social products

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

p1666009312888229/1665827449.379749-slack-C02JJ910CNL

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* On a JN site or your local, purchase the Jetpack Social Basic plan
* Without this PR, it won't show up under my products.
* After checking out to this branch, it will show up as shown in this image. 
Screenshot 2022-10-21 at 7 31 09 PM

### CODE REVIEW

**Comment by spsiddarthan** [comment]
I will fix the linting issues, the method doc comments that I didn't tweak got automatically added - I will have them removed

File: projects/plugins/jetpack/_inc/client/components/plans/plan-icon/index.jsx
Comment (pablinos) [inline projects/plugins/jetpack/_inc/client/components/plans/plan-icon/index.jsx]: Do we need Social icons here? We should be able to use a variation on the SVG icon we use elsewhere.

File: projects/plugins/jetpack/_inc/client/lib/plans/constants.js
Comment (pablinos) [inline projects/plugins/jetpack/_inc/client/lib/plans/constants.js]: Are we adding Premium here to prevent having to do this again in a few weeks? That's probably ok, as long as we can decide on the text, icon, slug name etc. Otherwise we'll need to revisit it anyway.

File: projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.jsx
Comment (pablinos) [inline projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.jsx]: This looks to be a copy of the Search card. We should update that. I'm not sure when this card is displayed, but we should test that it looks ok too.

File: projects/plugins/jetpack/_inc/client/my-plan/my-plan-header/index.js
Comment (pablinos) [inline projects/plugins/jetpack/_inc/client/my-plan/my-plan-header/index.js]: The other tag lines seem to be of the form 'You can...', 'Your site is...' - focusing an the features 'you' are benefiting from. We should probable rephrase that here, but I'm not sure what's best. "You can automatically share your content to social media sites.", maybe?

File: projects/plugins/jetpack/_inc/client/my-plan/my-plan-header/index.js
Comment (pablinos) [inline projects/plugins/jetpack/_inc/client/my-plan/my-plan-header/index.js]: This partly why I'm not sure if we should include premium at this stage. I don't think we're in a position to write this tag line yet.

**Review by pablinos — COMMENTED** [review]
This is working as described. I think there are a few bits we should look at before it's ready to merge.

**Comment by sdixon194** [comment]
@spsiddarthan Just a headsup that I think there are some comments that need addressing + the linting issues before we get this merged for the 11.5-beta tomorrow.

File: projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.jsx
Comment (pablinos) [inline projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.jsx]: I'm not sure this is ever the case, as it's not actually a plan at this point. I think we'd need to take the same approach as search and check for the feature.

There are other blocks where this `hasInstantSearch` check is made too, so we might have to add it in there as well

File: projects/plugins/jetpack/_inc/client/lib/plans/constants.js
Comment (spsiddarthan) [reply projects/plugins/jetpack/_inc/client/lib/plans/constants.js]: Yup, that's the reason. Didn't want to come back and wait for a release for after we launch premium, but we probably have another release between and now then, so perhaps we can exclude premium for now

File: projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.jsx
Comment (spsiddarthan) [reply projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.jsx]: I will do that!

File: projects/plugins/jetpack/_inc/client/lib/plans/constants.js
Comment (spsiddarthan) [reply projects/plugins/jetpack/_inc/client/lib/plans/constants.js]: You're right, I wanted to not wait for another release when we launch premium plans, but makes sense not to add something that's not actually ready yet and won't be for a bit.

**Review by pablinos — APPROVED** [review]
This is working well with me. We might want to get Crew to check it, seeing as I've worked on it too.

**Review by bindlegirl — APPROVED** [review]
LGTM

**Pull request merged**
